hack/update-helm-repo.sh

```diff
@@ -19,5 +19,8 @@ set -o pipefail
 
 REPO_ROOT=$(dirname "${BASH_SOURCE}")/..
 
-helm package ${REPO_ROOT}/helm/cloud-provider-azure -d ${REPO_ROOT}/helm/repo
-helm repo index ${REPO_ROOT}/helm/repo
+helm package ${REPO_ROOT}/helm/cloud-provider-azure -d ${REPO_ROOT}/helm/repo/new
```

Comment:
This PR also updates the helm repo foo so that we don't overwrite the `created` timestamp for a version every time we publish a new release.

See this thread:

https://github.com/helm/helm/issues/7363